src/array/primitive/mutable.rs

```diff
@@ -234,12 +234,6 @@ impl<T: NativeType> MutablePrimitiveArray<T> {
         let len = self.len();
         if let Some(validity) = self.validity.as_mut() {
             validity.extend_constant(len - validity.len(), true);
-        } else {
```

Comment:
There was not `validity` and we extend with `true` so we know that `null_count == 0` no need to compute it.